packages/engine/src/postprocessing/components/PostProcessingComponent.tsx

```diff
@@ -0,0 +1,572 @@
+  const renderer = useComponent(rendererEntity, RendererComponent)
+
+  let lut1DEffectTexturePath: string | undefined
+  if (
+    postprocessingComponent.effects[Effects.LUT1DEffect].lutPath &&
+    postprocessingComponent.effects[Effects.LUT1DEffect].isActive.value
+  ) {
+    lut1DEffectTexturePath = postprocessingComponent.effects[Effects.LUT1DEffect].lutPath.value
+  }
+  let lut3DEffectTexturePath: string | undefined
+  if (
+    postprocessingComponent.effects[Effects.LUT3DEffect].lutPath &&
+    postprocessingComponent.effects[Effects.LUT3DEffect].isActive.value
+  ) {
+    lut3DEffectTexturePath = postprocessingComponent.effects[Effects.LUT3DEffect].lutPath.value
+  }
+  let textureEffectTexturePath: string | undefined
+  if (
+    postprocessingComponent.effects[Effects.TextureEffect].texturePath &&
+    postprocessingComponent.effects[Effects.TextureEffect].isActive.value
+  ) {
+    textureEffectTexturePath = postprocessingComponent.effects[Effects.TextureEffect].texturePath.value
+  }
+
+  const [lut1DEffectTexture, lut1DEffectTextureError] = useTexture(lut1DEffectTexturePath!, entity)
+  const [lut3DEffectTexture, lut3DEffectTextureError] = useTexture(lut3DEffectTexturePath!, entity)
+  const [textureEffectTexture, textureEffectTextureError] = useTexture(textureEffectTexturePath!, entity)
+
+  const scene = useHookstate<Scene>(() => new Scene())
+  const normalPass = useHookstate<CustomNormalPass>(() => new CustomNormalPass(scene, camera))
+  const depthDownsamplingPass = useHookstate<DepthDownsamplingPass>(
+    () =>
+      new DepthDownsamplingPass({
+        normalBuffer: normalPass.value.texture,
+        resolutionScale: 0.5
+      })
+  )
+  const velocityDepthNormalPass = useHookstate(new VelocityDepthNormalPass(scene, camera))
+  const useVelocityDepthNormalPass = useHookstate(false)
+  const useDepthDownsamplingPass = useHookstate(false)
+  const renderMode = getState(RendererState).renderMode
+  const effects = useHookstate<Record<string, Effect>>({})
+
+  useEffect(() => {
+    const composer = new EffectComposer(renderer.value.renderer as WebGLRenderer)
+    renderer.effectComposer.set(composer)
+    const renderPass = new RenderPass()
+    renderer.value.effectComposer.addPass(renderPass)
+    renderer.renderPass.set(renderPass)
+  }, [])
+
+  useEffect(() => {
+    velocityDepthNormalPass.set(new VelocityDepthNormalPass(scene, camera))
+  }, [scene])
+
+  useEffect(() => {
+    const renderSettings = getState(RendererState)
+
+    const effectsVal = effects.get(NO_PROXY) as Record<string, Effect>
+
+    if (renderSettings.usePostProcessing) {
+      for (const key in effectsVal) {
+        const val = effectsVal[key]
+        renderer.value.effectComposer[key] = val
+      }
+    }
+
+    //always have the smaa effect
+    const smaaPreset = getState(RenderSettingsState).smaaPreset
+    const smaaEffect = new SMAAEffect({
+      preset: smaaPreset,
+      edgeDetectionMode: EdgeDetectionMode.COLOR
```

Comment:
seems like we forgot to add this to the effect composer. Gonna do it now

Reply:
Forgot the reason for this. A lot of the new effects are convolution effects that can't mix with SMAA. Gonna just expose in editor and allow people to decide at scene authoring I think. Though we gotta consider error messaging or handling for this library. It's a headache.